src/com/example/Demo_Subj/RoomView.java

```diff
@@ -7,68 +7,117 @@
+        World.setRoom(2, new Room(2, R.drawable.wohnzimmer, 0.0, 0.0, itemListWohn, this.ctx));
+    }
+
+    private void fillItemList(){                                                                //Existiert temporär, bis der XML-Parser eingebunden wurde
+        itemListWohn = new ArrayList<Integer>();
+        World.setItem(1, new Item(1, R.drawable.pflanze, GlobalInformation.getScreenWidth()*0.5, GlobalInformation.getScreenHeight()*0.5, null, null, null, null, null, ctx));
+        b &= itemListWohn.add(World.getItemById(1).getID());
+        World.setItem(2, new Item(2, R.drawable.boddle, GlobalInformation.getScreenWidth()*0.3, GlobalInformation.getScreenHeight()*0.5, null, null, null, null, null, ctx));
+        b &= itemListWohn.add(World.getItemById(2).getID());
+
+        itemListSchlaf = new ArrayList<Integer>();
+        World.setItem(3, new Item(3, R.drawable.boddle, GlobalInformation.getScreenWidth()*0.3, GlobalInformation.getScreenHeight()*0.5, null, null, null, null, null, ctx));
+        b &= itemListSchlaf.add(World.getItemById(3).getID());
     }
 
     @Override
     protected void onDraw(Canvas canvas){
         super.onDraw(canvas);
-
         // find out which room to draw
         //TODO should be replaced by a proper find() in the list
-        for (Iterator<Room> iter = roomList.iterator(); iter.hasNext(); ) {
-            Room room = iter.next();
-            if (room.getRoomID() == GlobalInformation.getCurrentRoom()) {
-                drawRoom = room;
-            }
+
+
+        if(lastRoom != GlobalInformation.getCurrentRoom()){                                     //hat sich lastroom gegenüber dem letzten Aufruf geändert? wenn ja dann lade den aktuellen Raum
+            drawRoom = World.getRoomById(GlobalInformation.getCurrentRoom());
+            drawRoomB = Bitmap.createScaledBitmap(
+                    BitmapFactory.decodeResource(resources, drawRoom.getPicResource()),
+                    GlobalInformation.getScreenWidth(),
+                    GlobalInformation.getScreenHeight(),
+                    false
+            );
+            lastRoom = drawRoom.getID();
         }
+
+
         if (drawRoom == null) {
             // room not found
-            drawRoom = roomList.get(0);
+            drawRoom = World.getRoomById(1);
         }
 
-        // draw the background image
-        canvas.drawBitmap(drawRoom.getBitmapRoom(), 0, 0, p);
+        canvas.drawBitmap(drawRoomB, 0, 0, p);
 
         // draw all items (TODO: add layering)
-        List<Item> drawItems = drawRoom.getItemList();
-        
-        for (Iterator<Item> iter = drawItems.iterator(); iter.hasNext(); ){
-            Item obj = iter.next();
-            canvas.drawBitmap(obj.getBitmapO(), obj.getxPos(), obj.getyPos(), p);
+        //Layer als zusätzliche Eigenschaft von Item benötigt, wird ein Offset zur Y-Koordinate von Objekten hinzufügen
+
+        int itemId;
+        if(b){
```

Comment:
Wie ist das mit diesem b hier? Welche Funktion hat das (der Name sagt es einem schonmal nicht^^)? So wie es scheint, willst du, dass Objekte nur eingefügt werden, wenn du vorhin welche erfolgreich in die Liste eingefügt hast. Rene wird aber mit dem XML-Parser bestimmt keine b-Variable setzen, dieser Code muss auch funktionieren (also abfangen), wenn die Liste z.B. leer ist, also keine Objekte im Raum sind.

Aber schön, dass du das auf die verwendung der get-Methoden umgeschrieben hast!